packages/utilities/src/functions/formatting.ts

```diff
@@ -101,3 +101,21 @@ export const buildCodemodSlug = (name: string) =>
     .replaceAll("@", "")
     .split(/[\/ ,.-]/)
     .join("-");
+
+export const extractNameAndVersion = (val: string) => {
+  const lastAtSymbolIndex = val.lastIndexOf("@");
+  const sliceUpToLastAtSymbol = val.substring(0, lastAtSymbolIndex);
+
+  // Starts with a @
+  if (sliceUpToLastAtSymbol.length === 0) {
+    return {
+      name: val,
+      version: null,
+    };
+  }
+
+  return {
+    name: sliceUpToLastAtSymbol,
+    version: val.substring(lastAtSymbolIndex + 1),
+  };
```

Comment:
how about using a regex here?

```ts
val.match(/^(?<name>@?[^@]+)(?:@(?<version>[0-9]+\.[0-9]+\.[0-9]+))?$/).groups
```

Reply:
i'd argue that overusing regex where it's unnecessary is damaging readability of the code and also might affect performance in more complex examples. i prefer to stay away from using regular expressions as much as possible.

Reply:
also version is not necessarily 3 digits separates by dots. we also support string versions such as "latest", "alpha" etc.